File: src/lowlevel/server/icefsoperators/icefsIoctl.go
/*
 * @Author: Tan90degrees tangentninetydegrees@gmail.com
 * @Date: 2023-03-11 07:18:32
 * @LastEditors: Tan90degrees tangentninetydegrees@gmail.com
 * @LastEditTime: 2023-04-18 04:08:11
 * @FilePath: /icefs/src/lowlevel/server/icefsoperators/icefsIoctl.go
 * @Description:
 *
 * Copyright (C) 2023 Tan90degrees <tangentninetydegrees@gmail.com>.
 */
package icefsoperators

import (
	"context"
	pb "icefs-server/icefsgrpc"
	"icefs-server/icefsthrift"
	"syscall"
)

func (s *IcefsServer) doIcefsIoctl(inode uint64, cmd uint32, arg []byte, flags uint32, inBufSize uint64, outBufSize uint64) (status int32) {
	return int32(syscall.ENOTSUP)
}

func (s *IcefsGRpcServer) DoIcefsIoctl(ctx context.Context, req *pb.IcefsIoctlReq) (*pb.IcefsIoctlRes, error) {
	var res pb.IcefsIoctlRes

	res.Status = s.server.doIcefsIoctl(req.Inode, req.Cmd, req.Arg, req.Flags, req.InBufSize, req.OutBufSize)

	return &res, nil
}

func (s *IcefsThriftServer) DoIcefsIoctl(ctx context.Context, req *icefsthrift.IcefsIoctlReq) (*icefsthrift.IcefsIoctlRes, error) {
	var res icefsthrift.IcefsIoctlRes

	res.Status = s.server.doIcefsIoctl(uint64(req.Inode), uint32(req.Cmd), req.Arg, uint32(req.Flags), uint64(req.InBufSize), uint64(req.OutBufSize))

	return &res, nil
}

File: src/lowlevel/client/icefsoperators/icefsIoctl.cpp
/*
 * @Author: Tan90degrees tangentninetydegrees@gmail.com
 * @Date: 2023-03-30 04:19:29
 * @LastEditors: Tan90degrees tangentninetydegrees@gmail.com
 * @LastEditTime: 2023-04-20 09:52:17
 * @FilePath: /icefs/src/lowlevel/client/icefsoperators/icefsIoctl.cpp
 * @Description:
 *
 * Copyright (C) 2023 Tan90degrees <tangentninetydegrees@gmail.com>.
 */
#include <stdio.h>

#include <string>

#include "icefsClient.hpp"
#include "icefsClientUtils.hpp"

void IcefsClient::DoIcefsIoctl(fuse_req_t fuseReq, fuse_ino_t inode,
                               unsigned int cmd, void *arg,
                               struct fuse_file_info *fi, unsigned flags,
                               const void *inBuf, size_t inBufSize,
                               size_t outBufSize) {
  ICEFS_PR_FUNCTION;

  fuse_reply_err(fuseReq, ENOTSUP);
}